scandium-core/src/main/java/org/eclipse/californium/scandium/util/SecretUtil.java

```diff
@@ -79,43 +80,48 @@ public static boolean isDestroyed(SecretKey key) {
 	}
 
 	/**
-	 * Check, if destroyable was destroyed.
+	 * Checks if a given destroyable has already been destroyed.
 	 * 
-	 * @param destroyable destroyable to check. If {@code null}, {@code true} is returned.
+	 * @param destroyable The destroyable to check (may be {@code null}).
+	 * @return {@code true} if the given object either is {@code null} or has been destroyed.
 	 */
 	public static boolean isDestroyed(Destroyable destroyable) {
 		return destroyable == null || destroyable.isDestroyed();
 	}
 
 	/**
-	 * Create secret key (with destroyable implementation).
+	 * Creates a secret key.
 	 * 
-	 * @param secret secret as byte array
-	 * @param algorithm algorithm the secret key is used for.
-	 * @return the secreate key
+	 * @param secret The key material.
+	 * @param algorithm The algorithm that the key is used for. The name
+	 *                  <em>PSK</em> should be used if the key is supposed to
+	 *                  be used with a PSK based TLS handshake.
+	 * @return The key.
 	 */
 	public static SecretKey create(byte[] secret, String algorithm) {
 		return new DestroyableSecretKeySpec(secret, algorithm);
 	}
 
 	/**
-	 * Create secret key (with destroyable implementation).
+	 * Create a secret key.
```

Comment:
Why removing the information about the key is destroyable ?

Reply:
No harm intended, it's just that `SecretKey` extends `Destroyable`, so IMHO it is obvious that the returned object implements `Destroyable`. Or am I missing something?

Reply:
`SecretKey` extends `Destroyable` since java 8 only, so as we aims java7 maybe it is worth to be clear about that.
(but I have clearly not a strong opinion on this)

Reply:
My bad, I didn't realize that and I keep forgetting that we are still Java 7 ...
I agree that we should document this explicitly. I will change it accordingly. Thanks for spotting :+1:

Reply:
@sbernard31 @boaks why don't we move `DestroyableSecretKeySpec` into its own class file and return that instead of returning a `SecretKey`? Then everybody knows what he gets back from the factory methods ...

Reply:
Maybe one argument could be that generally this is better for a factory to return interface instead of class. :thinking: 

My opinion : both are OK for me !

(Not directly linked but maybe in the javadoc we can add that the key destroy() SHOULD be called after key usage)

Reply:
Even if it's not implemented for now, but using e.g. android HSM support, will return a `SecretKey`and not my internal, temporary work-around "DestroyableSecretKeySpec`implementation. So I would prefer to keep the types as they are.

Reply:
@boaks @sbernard31 Then what about an interface `DestroyableSecretKey` which extends `SecretKey` and `Destroyable` and returning that instead? The current implementation can then return `DestroyableSecretKeySpec` (which should then implement `DestroyableSecretKey`) and later, when you also support Android HSM you wrap the object that you get from Android appropriately before returning it. WDYT?

Reply:
Most are not longer using java 7. That "work-around" is mainly caused by the very delayed release. For me, it's not really worth to introduce something, which the most will wonder about, because it try to fix something in an deprecated java version. I only did it, to still comply to the java 7 promise.